app/services/email_confirmation_token_validator.rb

```diff
@@ -29,7 +29,9 @@ def email_address_already_confirmed_by_user?(user)
 
   def confirmation_period_expired?
     return false unless email_address_found_with_token?
-    expiration_time = @email_address.confirmation_sent_at + email_valid_for_duration
+    confirmation_sent_at = @email_address.confirmation_sent_at
```

Comment:
How is `confirmation_sent_at` ever blank? Don't we send the email as soon as we create the email address record?

Reply:
I have no idea.  I don't see it anywhere in the code (token is usually nil'd out with it).  Is it possible this is leftover from a past migration of data?

Reply:
Found the issue.  A nil confirmation token was being sent (ie no param) and was actually being found in the db.